src/main/java/org/moltimate/moltimatebackend/alignment/AlignmentUtils.java

```diff
@@ -0,0 +1,77 @@
+package org.moltimate.moltimatebackend.alignment;
+
+import org.biojava.nbio.structure.Group;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+public class AlignmentUtils {
+
+    //ALA, CYS, ASP, GLU, PHE, GLY, HIS, ILE, LYS, LEU, MET, ASN, PRO, GLN, ARG, SER, THR, VAL, TRP, TYR
+
+    public static final HashMap aminoAcidToCharMap = new HashMap<String, String>() {{
+        put("ALA", "A");
+        put("CYS", "B");
+        put("ASP", "C");
+        put("GLU", "D");
+        put("PHE", "E");
+        put("GLY", "F");
+        put("HIS", "G");
+        put("ILE", "H");
+        put("LYS", "I");
+        put("LEU", "J");
+        put("MET", "K");
+        put("ASN", "L");
+        put("PRO", "M");
+        put("GLN", "N");
+        put("ARG", "O");
+        put("SER", "P");
+        put("THR", "Q");
+        put("VAL", "R");
+        put("TRP", "S");
+        put("TYR", "T");
+    }};
+
+    public static String groupListToResString(List<Group> residues) {
+        StringBuilder stringBuilder = new StringBuilder();
+        residues.forEach(residue -> stringBuilder.append(aminoAcidToCharMap.get(residue.getChemComp().getThree_letter_code().toUpperCase())));
+        return stringBuilder.toString();
+    }
+
+    public static String residueListToResString(List<String> resList) {
+        StringBuilder stringBuilder = new StringBuilder();
+        resList.forEach(residue -> stringBuilder.append(aminoAcidToCharMap.get(residue)));
+        return stringBuilder.toString();
+    }
+
+    public static int levensteinDistance(String x, String y) {
+        int[][] dp = new int[x.length() + 1][y.length() + 1];
+
+        for (int i = 0; i <= x.length(); i++) {
+            for (int j = 0; j <= y.length(); j++) {
+                if (i == 0) {
+                    dp[i][j] = j;
+                } else if (j == 0) {
+                    dp[i][j] = i;
+                } else {
+                    dp[i][j] = min(dp[i - 1][j - 1]
+                                    + costOfSubstitution(x.charAt(i - 1), y.charAt(j - 1)),
+                            dp[i - 1][j] + 1,
+                            dp[i][j - 1] + 1);
+                }
+            }
+        }
+
+        return dp[x.length()][y.length()];
+    }
+
+    private static int costOfSubstitution(char a, char b) {
+        return a == b ? 0 : 1;
+    }
+
+    private static int min(int... numbers) {
+        return Arrays.stream(numbers)
+                .min().orElse(Integer.MAX_VALUE);
```

Comment:
Newlines for the method call daisy chain

Reply:
DOne